api/user/posts/posts.py

```diff
@@ -1,46 +1,220 @@
-    """update the post whose id is POST_ID (must be authenticated)"""
-    pass
+    """
+    Update the post whose id is POST_ID (must be authenticated) ie the person
+    editing the post must be the author of the post.
+    json received must have all columns to be updated as key value pairs
+    """
+    # TODO handle authentication
+    json = request.json
+
+    author = Author.query.filter_by(object_id=author_id).first()
+    update_stmt = (
+        update(Post)
+        .where(object_id=post_id, author_id=author.id, inbox=author_id)
+        .values(**json)
+        .execution_options(synchronize_session="fetch")
+    )
+
+    db.session.execute(update_stmt)
+    db.session.commit()
 
 
 @posts_bp.route("/<string:author_id>/posts/<string:post_id>", methods=["DELETE"])
 def delete_post(author_id: str, post_id: str):
     """remove the post whose id is post_id"""
-    pass
+    author = Author.query.filter_by(object_id=author_id).first()
+    del_stmt = (
+        delete(Post)
+        .where(object_id=post_id, author_id=author.id, inbox=author_id)
+        .execution_options(synchronize_session="fetch")
+    )
+
+    db.session.execute(del_stmt)
+    db.session.commit()
 
 
 @posts_bp.route("/<string:author_id>/posts/<string:post_id>", methods=["PUT"])
 def create_post(author_id: str, post_id: str):
+    # TODO is this a new post or does it have existing comments???
     """create a post where its id is post_id"""
-    pass
+    response = {}
+    data = request.json
+
+    data = request.json
+    published = data.get("published")
+    title = data.get("title")
+    origin = data.get("origin")
+    source = data.get("source")
+    description = data.get("description", "")
+    contentType = data.get("contentType")
+    content = data.get("content")
+    categories = ",".join(data.get("categories"))
+    visibility = data.get("visibility")
+    unlisted = data.get("unlisted")
+    author_url = ""
+    try:
+        author = Author.query.filter_by(object_id=author_id).first()
+        post = Post(
+            id=post_id,
+            published=published,
+            title=title,
+            origin=origin,
+            source=source,
+            description=description,
+            content=content,
+            contentType=contentType,
+            categories=categories,
+            visibility=visibility,
+            unlisted=unlisted,
+            author_id=author_url,
+            inbox=author_id,
+        )
+
+        db.session.add(post)
+        db.session.commit()
+
+    except:
+        response["Success"] = 0
+        response["Message"] = "Author does not exist on this server."
+    else:
+        author_url = author.id
+        response["Success"] = 1
+        response["Message"] = "Post created succesfully."
+
+    return response
 
 
 @posts_bp.route("/<string:author_id>/posts", methods=["POST"])
 def create_post_auto_gen_id(author_id: str):
     """create a new post but generate a new id"""
-    pass
+    response = {}
```

Comment:
This is very very close to the endpoint above, could you extract the common functionality to a function and then just call that function?